src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -7926,6 +8099,70 @@ private bool EmitLHSThisDotProperyForBinaryExpr(AssociativeNode bnode, ref Proto
             return false;
         }
 
+        private void HandlePointerList(BinaryExpressionNode bnode)
+        {
+            // All associative code is SSA'd and we want to keep track of the original identifier nodes of an identifier list:
+            //      i.e. x.y.z
+            // These identifiers will be used to populate the real graph nodes dependencies
+            if (bnode.isSSAPointerAssignment)
+            {
+                Validity.Assert(null != stackSSAPointerList);
+
+                if (bnode.IsFirstIdentListNode)
+                {
+                    stackSSAPointerList.Push(new List<AssociativeNode>());
+                }
+
+                if (bnode.RightNode is IdentifierNode)
```

Comment:
Sure, the AST replacer is certainly a cleaner approach. 
For this I will have to explore using the AstReplacer for the SSA transform.
The SSAIdentList is only one small part of BuildSSA(). I want to make sure that the entire SSA transform algorithm is refactored to use the applicable patterns.

Pls see: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7435